component/loki/source/file/file.go

```diff
@@ -36,9 +37,32 @@ const (
 // Arguments holds values which are used to configure the loki.source.file
 // component.
 type Arguments struct {
-	Targets   []discovery.Target  `river:"targets,attr"`
-	ForwardTo []loki.LogsReceiver `river:"forward_to,attr"`
-	Encoding  string              `river:"encoding,attr,optional"`
+	Targets             []discovery.Target  `river:"targets,attr"`
+	ForwardTo           []loki.LogsReceiver `river:"forward_to,attr"`
+	Encoding            string              `river:"encoding,attr,optional"`
+	DecompressionConfig DecompressionConfig `river:"decompression,block,optional"`
+}
+
+type DecompressionConfig struct {
+	Enabled      bool          `river:"enabled,attr"`
+	InitialDelay time.Duration `river:"initial_delay,attr,optional"`
+	Format       string        `river:"format,attr"`
+}
+
+func (d DecompressionConfig) Validate() error {
+	if d.Format == "" {
+		return nil
+	}
+	for sf := range supportedCompressedFormats() {
+		if d.Format == sf {
+			return nil
+		}
+	}
+	return fmt.Errorf(
+		"unsupported compression format: %q - please use one of %q",
+		d.Format,
+		strings.Join(maps.Keys(supportedCompressedFormats()), ", "),
+	)
```

Comment:
Added this to eagerly find errors and give feedback:
![image](https://github.com/grafana/agent/assets/17101802/945882a3-8c6d-4c7a-af0e-fa676bd4dcfe)